src/Cli/dotnet/commands/RestoringCommand.cs

```diff
@@ -53,14 +53,17 @@ public class RestoringCommand : MSBuildForwardingApp
                 return null;
             }
 
-            var restoreArguments = parsedArguments
-                .Where(a => !IsExcludedFromRestore(a))
-                .Concat(trailingArguments);
+            IEnumerable<string> restoreArguments = new string[] { "-target:Restore" };
```

Comment:
the existing logic does not have "-target:Restore". And usually ` msbuild -target:Restore;Build` is a bug since restore will include more props. `msbuild /restore -target:build` is the right way

Reply:
This logic existed before, just in a different location. Before, we called RestoreCommand.FromArgs (see line 60) to get the restore command whereas now we're using the RestoreCommand constructor directly (because we don't want to have to re-parse everything). The FromArgs method adds "-target:Restore" to the arguments whereas the constructor doesn't.